torch/nn/modules/module.py

```diff
@@ -2037,6 +2037,11 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
                                       'whose dimensions in the checkpoint are {}, '
                                       'an exception occurred : {}.'
                                       .format(key, param.size(), input_param.size(), ex.args))
+                    if (str(ex) == "Cannot copy out of meta tensor; no data!"):
+                        error_msgs.append('Did you mean to pass `assign=True` to '
```

Comment:
Another way to message here would be to to say: copying into a meta Tensor is not possible. If you are skipping initialization, you should be setting assign=True when loading the state dict.
I think it's fine to assume that the user ended up here because they're skipping init.

Reply:
Hm the case in the issue was that the user was copying from meta tensor in the state dict into a real tensor in the model, which is not really the skip init case I think

Reply:
Ho. How did they get a meta state dict though? :thinking: 

Detecting copy of a real Tensor into a meta one sounds like a good thing to warn about indeed!